include/swift/AST/Identifier.h

```diff
@@ -386,6 +403,14 @@ class DeclName {
     return lhs.compare(lhs) >= 0;
   }
 
+  bool operator ==(StringRef other) {
```

Comment:
Please do not add these members. Anything that operates only on the base name wouldn't be a correct `==` anyway.